lib/semantics/resolve-names.cc

```diff
@@ -4850,6 +4847,146 @@ bool ResolveNamesVisitor::BeginScope(const ProgramTree &node) {
   }
 }
 
+// Perform checks that need to happen after all of the specification parts
+// but before any of the execution parts.
+void ResolveNamesVisitor::FinishSpecificationParts(const ProgramTree &node) {
+  if (!node.scope()) {
+    return;  // error occurred creating scope
```

Comment:
Should this be an assert or are we expecting scope creation errors?

Reply:
We can fail to create a scope, for example, if a submodule names a parent module that doesn't exist. There is no good way to create a meaningful scope in that case.